stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java

```diff
@@ -260,6 +260,8 @@ public void run() {
         LogUtil.w("I/O error: %s", ex);
       } catch (HttpException ex) {
         LogUtil.w("Unrecoverable HTTP protocol violation: %s", ex);
+      } catch (OutOfMemoryError ex) {
```

Comment:
Way too high level of a catch.  Strongly recommend putting this in two critical places, `Network#readResponseBody` (specifically around the `readFile` call) and also `NetworkEventReporterImpl#readBodyAsString`.

I prefer this strategy because it is very specific about suppressing this serious error only in call sites we expect could accidentally allocate much larger segments than we normally expect.  No other code should be doing this.  Also, catching it appropriate lower levels allows us to respond using the actual DevTools protocol so that the UI can represent the error in a discoverable fashion.